5.5/docker-entrypoint.sh

```diff
@@ -58,6 +58,11 @@ _datadir() {
 	"$@" --verbose --help --log-bin-index="$(mktemp -u)" 2>/dev/null | awk '$1 == "datadir" { print $2; exit }'
 }
 
+# For use with the client if user passes the --socket argument
+_socket() {
+	"$@" --verbose --help --log-bin-index="$(mktemp -u)" 2>/dev/null | awk '$1 == "socket" { print $2; exit }'
+}
```

Comment:
Do you think there'd be any value in making something like a `_get_config` function to normalize these two?  I guess `_check_config` is pretty similar too.

Something like this is what I was kind of thinking:
```bash
_get_config() {
	local conf="$1"; shift
	"$@" --verbose --help --log-bin-index="$(mktemp -u)" 2>/dev/null | awk '$1 == "'"$conf"'" { print $2; exit }'
}

datadir="$(_get_config 'datadir' "$@")"
socket="$(_get_config 'socket' "$@")"
```

(will defer to your decision, just wanted to throw the idea out there in case you hadn't considered it :+1:)